examples/js/pickupable.component.js

```diff
@@ -0,0 +1,33 @@
+AFRAME.registerComponent('pickupable', {
+  init: function() {
+    var self = this;
+    self.player = document.querySelector('#player');  //this is our player/camera
+    self.pickedUp = false;
+
+    //let's give a random material color that syncs appropriately
+    self.el.setAttribute('material', {flatShading:true, color:'rgb(' + Math.floor(Math.random() * 255) + ',' + Math.floor(Math.random() * 255) + ',' + Math.floor(Math.random() * 255) + ')'});
+
+    self.el.addEventListener('click', (e) => {
+        if (self.pickedUp === true) {
+            //release
+            self.el.sceneEl.object3D.attach(self.el.object3D); //using three's "attch" allows us to retain world transforms during pickup/release
+            self.pickedUp = false;
+        }
+        else {
+            //pick-up
+            self.player.object3D.attach(self.el.object3D);
```

Comment:
Why do you attach it the the camera in the first place? The whole PR exists only because of it.
You shouldn't need to detach and attach the object when it's grabbed.
You should look at the https://github.com/wmurphyrd/aframe-super-hands-component library instead.